server/controllers/userController.js

```diff
@@ -0,0 +1,82 @@
+const { User } = require('../models/index');
+
+const getAllUsers = async (req, res) => {
+    try {
+        const users = await User.findAll();
+        if (!users) {
+            res.status(404).send("Error fetching users: Users not found");
+        } else {
+            res.status(200).json(users);
+        }
+    } catch (error) {
+        res.status(500).send(error);
+    }
+}
+
+const getUser = async (req, res) => {
+    try {
+        const userId = req.params.userId
+        const user = await User.findByPk(userId);
+        if (!user) {
+            res.status(404).send(`Error fetching user: User with ID ${userId} not found`);
+        } else {
+            res.status(200).json(user);
+        }
+    } catch (error) {
+        res.status(500).send(error);
+    }
+}
+
+const createUser = async (req, res) => {
+    try {
+        const userInfo = req.body;
+        await User.create(userInfo);
+        const newUser = await User.findOne({ where: { email: userInfo.email } });
```

Comment:
I think it makes more sense to save   -await User.create(userInfo)- into a variable and then check if that variable's value is null.
Here you are querying db 2 times, first time you are sending POST req to create a user, and then the second query to check of you can find the user in the database, imo it's unnecessary db call we can avoid